test/integration/crud/bulk.test.ts

```diff
@@ -717,507 +690,415 @@ describe('Bulk', function () {
-            const error = result.getWriteErrorAt(0);
-            expect(error.code === 11000).to.equal(true);
-            expect(error.errmsg).to.exist;
-            expect(err.message).to.equal(error.errmsg);
-
-            client.close(done);
-          });
-        }
-      );
-    });
+    // Initialize the unordered Batch
+    const batch = col.initializeUnorderedBulkOp();
+
+    // Add some operations to be executed in order
+    batch.insert({ a: 1 });
+    batch.insert({ a: 1 });
+
+    // Execute the operations
+    const error = await batch.execute(configuration.writeConcernMax()).catch(err => err);
+    // Test basic settings
+    const result = error.result;
+    expect(result.insertedCount).to.equal(1);
+    expect(result.hasWriteErrors()).to.equal(true);
+    expect(result.getWriteErrorCount() === 1).to.equal(true);
+
+    // Individual error checking
+    const writeError = result.getWriteErrorAt(0);
+    expect(writeError.code === 11000).to.equal(true);
+    expect(writeError.errmsg).to.exist;
+    expect(error.message).to.equal(writeError.errmsg);
   });
 
   it(
     'should Correctly Execute Unordered Batch of with upserts causing duplicate key errors on updates',
     {
-      metadata: { requires: { topology: ['single', 'replicaset', 'ssl', 'heap', 'wiredtiger'] } },
+      metadata: { requires: { topology: ['single', 'replicaset'] } },
 
-      test: function (done) {
-        client.connect((err, client) => {
-          const db = client.db();
-          const col = db.collection('batch_write_unordered_ops_legacy_7');
-
-          // Add unique index on b field causing all updates to fail
-          col.createIndex({ b: 1 }, { unique: true, sparse: false }, err => {
-            expect(err).to.not.exist;
-
-            // Initialize the unordered Batch
-            const batch = col.initializeUnorderedBulkOp();
-
-            // Add some operations to be executed in order
-            batch.insert({ a: 1 });
-            batch.find({ a: 1 }).update({ $set: { b: 1 } });
-            batch
-              .find({ a: 2 })
-              .upsert()
-              .update({ $set: { b: 2 } });
-            batch
-              .find({ a: 3 })
-              .upsert()
-              .update({ $set: { b: 3 } });
-            batch.find({ a: 1 }).update({ $set: { b: 1 } });
-            batch.insert({ b: 1 });
-
-            // Execute the operations
-            batch.execute({}, function (err, result) {
-              expect(err).to.exist;
-              expect(result).to.not.exist;
-
-              // Test basic settings
-              result = err.result;
-              test.equal(2, result.insertedCount);
-              test.equal(2, result.upsertedCount);
-              test.ok(0 === result.modifiedCount || result.modifiedCount == null);
-              test.equal(0, result.deletedCount);
-              test.equal(true, result.hasWriteErrors());
-              test.ok(1, result.getWriteErrorCount());
```

Comment:
The same here, `test.ok` is not validating count of write errors. In this test setup actual number is 2, both on setting `b: 1`.